### PR TITLE
Feature/application containerization

### PR DESCRIPTION
**Pull request by HattoriHenzo**
@Vishal1297 Add docker file and docker compose file. And little refactoring.
I am planning unit tests if needed

### CODE REVIEW

**Comment by Vishal1297** [comment]
@HattoriHenzo 

Did you test docker compose ?

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,4 @@
+FROM adoptopenjdk/openjdk11:alpine-jre
```
Comment (Vishal1297) [inline Dockerfile:1]: Since project is using java 8 so set base image for java 8.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+WORKDIR /opt/app
```
Comment (Vishal1297) [inline Dockerfile:2]: Choosing root of file system will be more simpler.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+COPY build/libs/ContactsManager-0.0.1-SNAPSHOT.jar ContactsManager-0.0.1.jar
+CMD ["java", "-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=*:5005", "-jar", "ContactsManager-0.0.1.jar"]
```
Comment (Vishal1297) [inline Dockerfile:4]: Expose port 8080 & use entry point to stop parameters overriding through command line.

**Review by Vishal1297 — CHANGES_REQUESTED** [review]
Some minor changes & testing is required.

@HattoriHenzo

File: Dockerfile (hunk 1 above)
Comment (HattoriHenzo) [reply Dockerfile:1]: Ok I will check it. Are you planning to move to Java 11?

Comment (Vishal1297) [reply Dockerfile:1]: No, for now Java 8 🚀  is good.

File: Dockerfile (hunk 2 above)
Comment (HattoriHenzo) [reply Dockerfile:2]: It is a good practice to put your jar file in a folder. And generaly the folder used is _/opt/app_.  Check the [adoptopenjdk ](https://hub.docker.com/r/adoptopenjdk/openjdk8) page.
I would suggest you to keep it like this.

Comment (Vishal1297) [reply Dockerfile:2]: I know, so no need to change just make sure everything is working.

File: Dockerfile (hunk 3 above)
Comment (Vishal1297) [reply Dockerfile:4]: @HattoriHenzo

> Did  you need more time or i complete this ??

Comment (HattoriHenzo) [reply Dockerfile:4]: I will post a new PR very soon

Comment (Vishal1297) [reply Dockerfile:4]: @HattoriHenzo then, i am closing this one.

**Pull request closed without merging**